test/core/security/grpc_tls_certificate_provider_test.cc

```diff
@@ -495,6 +495,43 @@ TEST_F(GrpcTlsCertificateProviderTest,
   CancelWatch(watcher_state_1);
 }
 
+TEST_F(GrpcTlsCertificateProviderTest,
+       FileWatcherCertificateProviderWithInvalidCertificateKeyPairOnRefresh) {
```

Comment:
Is this the only test case you plan on adding?

Reply:
No. I will be adding a test for loading a mismatching certificate-key pair on creation. I have to talk with Zhen to clarify the expected behavior